internal/commands/vaultsecrets/run/run.go

```diff
@@ -0,0 +1,186 @@
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Inject secrets as environment variables:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets run 'env'
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "COMMAND",
+					Documentation: "Defines the invocation of the child process to inject secrets to.",
+				},
+			},
+			Validate: cmd.ArbitraryArgs,
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "app",
+					DisplayValue: "NAME",
+					Description:  "The application you want to pull all secrets from.",
+					Value:        appname.Flag(),
+				},
+			},
+		},
+		PersistentPreRun: func(c *cmd.Command, args []string) error {
+			return appname.Require(ctx)
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.Command = args[0:]
+			opts.AppName = appname.Get()
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRun(opts)
+		},
+	}
+	for _, f := range cmd.Flags.Local {
+		if f.Name == "app" {
+			f.Autocomplete = helper.PredictAppName(ctx, cmd, opts.PreviewClient)
+		}
+	}
+
+	return cmd
+}
+
+func runRun(opts *RunOpts) (err error) {
+	if len(opts.Command) == 0 {
+		return fmt.Errorf("failed to run app secrets from name %q - no command provided", opts.AppName)
+	}
+
+	envSecrets, err := getAllSecretsForEnv(opts)
+	if err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.AppName, err)
+	}
+
+	childProcess := setupChildProcess(opts.Ctx, opts.Command, envSecrets)
+
+	err = childProcess.Run()
+	if err != nil {
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			exitCode := exitErr.ExitCode()
+			return fmt.Errorf("failed to run with secrets in app %q; received code %v : %w", opts.AppName, exitCode, err)
+		} else {
+			return fmt.Errorf("failed to run with secrets in app %q: %w", opts.AppName, err)
+		}
+	}
+
+	return nil
+}
+
+func getAllSecretsForEnv(opts *RunOpts) ([]string, error) {
+	params := preview_secret_service.NewOpenAppSecretsParamsWithContext(opts.Ctx)
+	params.OrganizationID = opts.Profile.OrganizationID
+	params.ProjectID = opts.Profile.ProjectID
+	params.AppName = opts.AppName
+
+	res, err := opts.PreviewClient.OpenAppSecrets(params, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	result := os.Environ()
+
+	for _, secret := range res.Payload.Secrets {
+		// we need to append results in case of duplicates we want secrets to override
+		switch {
+		case secret.RotatingVersion != nil:
+			for name, value := range secret.RotatingVersion.Values {
+				result = append(result, fmt.Sprintf("%v_%v=%v", strings.ToUpper(secret.Name), name, value))
+			}
+		case secret.StaticVersion != nil:
+			result = append(result, fmt.Sprintf("%v=%v", strings.ToUpper(secret.Name), secret.StaticVersion.Value))
+		}
+	}
+
+	return result, nil
+}
+
+func setupChildProcess(ctx context.Context, command []string, envVars []string) *exec.Cmd {
```

Comment:
Testing this function directly would be good to ensure you are parsing the command and setting the cmd up as expected.